src/clients/HubPoolClient.ts

```diff
@@ -646,11 +643,43 @@ export class HubPoolClient {
     this.disputedRootBundles.push(
       ...events["RootBundleDisputed"].map((event) => spreadEventWithBlockNumber(event) as DisputedRootBundle)
     );
-    this.executedRootBundles.push(
-      ...events["RootBundleExecuted"]
-        .filter((event) => !this.configOverride.ignoredHubExecutedBundles.includes(event.blockNumber))
-        .map((event) => spreadEventWithBlockNumber(event) as ExecutedRootBundle)
-    );
+
+    const configStoreVersions: { [blockNumber: number]: number } = {};
+    for (const event of events["RootBundleExecuted"]) {
+      if (this.configOverride.ignoredHubExecutedBundles.includes(event.blockNumber)) {
+        continue;
+      }
+
+      if (!isDefined(configStoreVersions[event.blockNumber])) {
+        const timestamp = (await event.getBlock()).timestamp;
```

Comment:
These are both really good points, and I'd overlooked how inefficient this is.

I re-evaluated and realised that we only lack a helper method in the `ConfigStoreClient` to be able to resolve the version based on the block number instead of the timestamp. I've proposed that [here](https://github.com/across-protocol/sdk-v2/pull/230), and have made the corresponding change in this PR [here](https://github.com/across-protocol/sdk-v2/pull/213/commits/c7860aba405043cd21f329a7d116c8869989e053).

Reply:
Awesome, thanks!